### PR TITLE
Add missing lambda and api gateway log retention

### LINKED ISSUE #107: Add cloudwatch log group retention
We have some log groups without retention for this lambda
```
populate-provider-versions
```

and for API gw

### PR DESCRIPTION
**Pull request by eranelbaz**
Fix #107 
use `aws_cloudwatch_log_group` resource to manage it 
As docs say
- [aws_api_gateway_stage | Resources | hashicorp/aws | Terraform | Terraform Registry](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_stage) 
- [aws_lambda_function | Resources | hashicorp/aws | Terraform | Terraform Registry](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function)

### CODE REVIEW

File: lambda.tf [hunk 1]
```diff
@@ -58,30 +59,30 @@ resource "aws_lambda_function" "api_function" {
 
   environment {
     variables = {
-      GITHUB_TOKEN_SECRET_ASM_NAME             = aws_secretsmanager_secret.github_api_token.name
-      PROVIDER_NAMESPACE_REDIRECTS             = jsonencode(var.provider_namespace_redirects)
-      PROVIDER_VERSIONS_TABLE_NAME             = aws_dynamodb_table.provider_versions.name
+      GITHUB_TOKEN_SECRET_ASM_NAME = aws_secretsmanager_secret.github_api_token.name
```
Comment (Yantrio) [inline lambda.tf]: can you revert this formatting change please

File: lambda.tf [hunk 2]
```diff
@@ -58,30 +59,30 @@ resource "aws_lambda_function" "api_function" {
...
 // create the lambda function from zip file
 resource "aws_lambda_function" "populate_provider_versions_function" {
+  depends_on = [aws_cloudwatch_log_group.populate_provider_versions_function_log_group]
   function_name = "${replace(var.domain_name, ".", "-")}-populate-provider-versions"
   description   = "A basic lambda to handle populating provider versions in dynamodb"
   role          = aws_iam_role.lambda.arn
   handler       = "populate-provider-versions"
   memory_size   = 128
-  timeout       = 10 * 60
+  timeout       = 60
```
Comment (Yantrio) [inline lambda.tf]: this timeout should be 10 minutes, not 60 seconds

File: lambda.tf [hunk 3]
```diff
@@ -58,30 +59,30 @@ resource "aws_lambda_function" "api_function" {
...
       POPULATE_PROVIDER_VERSIONS_FUNCTION_NAME = aws_lambda_function.populate_provider_versions_function.function_name
-      GITHUB_API_GW_URL                        = var.domain_name
     }
   }
 }
 
 // ensure we have provisioned concurrency for the lambda function
 resource "aws_lambda_provisioned_concurrency_config" "api_function" {
-  function_name                     = aws_lambda_function.api_function.function_name
+  function_name = aws_lambda_function.api_function.function_name
```
Comment (Yantrio) [inline lambda.tf]: revert this formatting change please.

**Review by Yantrio — CHANGES_REQUESTED** [review]
Overall this looks good, just a couple nitpicks.

File: lambda.tf (hunk 1 above)
Comment (eranelbaz) [reply lambda.tf]: dfc4e976(rollback formatting)

File: lambda.tf (hunk 2 above)
Comment (eranelbaz) [reply lambda.tf]: dfc4e976(rollback formatting)

File: lambda.tf (hunk 3 above)
Comment (eranelbaz) [reply lambda.tf]: dfc4e976(rollback formatting)

**Comment by eranelbaz** [comment]
@Yantrio rollback those formatting issues + timeouts

**Review by Yantrio — APPROVED** [review]

**Pull request merged**